### PR TITLE
[add] webpack to repo

### PR DESCRIPTION
**Pull request by Way2nnadi**
Major Changes

- Now using webpack v4 to build react code
   - used for loading JS, CSS/SASS and images
- Now using babel to compile Javascript code
- Unified the `package.json`
- new commands 
    - `npm run dev`: used during development
    - `npm run build`: used for production
- cleaned up some of the code

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const { connectToDB } = require('./helpers/db');
 
 const app = express();
 const server = http.createServer(app);
-const port = process.env.PORT || 4000;
+const { PORT = 4000 } = process.env;
```
Comment (Way2nnadi) [inline server.js]: Can anyone tell me what is happening on this line?

Comment (Simpcyclassy) [reply server.js]: I think you used the ES6 shorter, cleaner and more understandable version of Node.js

Comment (EngrEric) [reply server.js]: Normally i think this is how the process object looks like.
process = {
      env: { 
          PORT: default_value_if_there_is
      }
   }
So, const { PORT = 4000 } = process.env;
will set PORT to 4000 even if there is something in the environment variable PORT.
In general i think the principle applied here is destructuring assignment syntax

File: server.js [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 
 app.use('/products', product);
 app.use('/producers', producer);
 
-const port = process.env.PORT || 4000;
-app.listen(port, () => console.log(`Listening on port ${port}`));
+server.listen(port, async() => {
+  await connectToDB();
```
Comment (EngrEric) [inline server.js]: why do you use async function here?

File: helpers/db.js [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 
-const db = mongoose.connection;
-db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+// Set up mongoose connection
+const connectToDB = () => {
+    mongoose.connect(MONGO_DB_URL, { useNewUrlParser: true }, () =>
+        console.error.bind(console, 'MongoDB connection error:')
+    );
+};
+
```
Comment (EngrEric) [inline helpers/db.js]: nice one sir

File: routes/productRoute.js [hunk 4]
```diff
@@ -1,25 +1,24 @@
...
-
-const { productCreate,
+const {
+  productCreate,
   productDetails,
   allProductsDetails,
   productDelete,
-  productUpdate } = productsController;
+  productUpdate,
+} = require('../controllers/productsController');
```
Comment (EngrEric) [inline routes/productRoute.js]: apart from shortening the number of codes, does it has more benefits

File: server.js (hunk 1 above)
Comment (EngrEric) [reply server.js]: Wow, that's superb.
so it does exactly same as this `const port = process.env.PORT || 4000;`

Comment (Way2nnadi) [reply server.js]: It does almost the same thing. `const { PORT = 4000 } = process.env;` sets a default value of 4000 on `PORT` only if `PORT` is `undefined`. Whereas `const port = process.env.PORT || 4000;` sets port to 4000 if `PORT` is null, false, 0, ""/'', or undefined.

**Pull request merged**